nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/classloader/ClassLoaderUtils.java

```diff
@@ -137,25 +137,17 @@ private static boolean isNotBlank(final String value) {
         return additionalClasspath.toArray(new URL[additionalClasspath.size()]);
     }
 
-    public static String generateAdditionalUrlsFingerprint(Set<URL> urls) {
-        List<String> listOfUrls = urls.stream().map(Object::toString).collect(Collectors.toList());
-        StringBuffer urlBuffer = new StringBuffer();
+    public static String generateAdditionalUrlsFingerprint(final Set<URL> urls) {
+        final StringBuilder fingerprintBuilder = new StringBuilder();
 
         //Sorting so that the order is maintained for generating the fingerprint
-        Collections.sort(listOfUrls);
-        try {
-            MessageDigest md = MessageDigest.getInstance("MD5");
-            listOfUrls.forEach(url -> {
-                urlBuffer.append(url).append("-").append(getLastModified(url)).append(";");
-            });
-            byte[] bytesOfAdditionalUrls = urlBuffer.toString().getBytes(StandardCharsets.UTF_8);
-            byte[] bytesOfDigest = md.digest(bytesOfAdditionalUrls);
-
-            return DatatypeConverter.printHexBinary(bytesOfDigest);
-        } catch (NoSuchAlgorithmException e) {
-            LOGGER.error("Unable to generate fingerprint for the provided additional resources {}", new Object[]{urls, e});
-            return null;
-        }
+        final List<String> sortedUrls = urls.stream().map(Object::toString).collect(Collectors.toList());
+        Collections.sort(sortedUrls);
```

Comment:
I think we can just use the `sorted()` on the stream.
```suggestion
        final List<String> sortedUrls = urls.stream()
            .map(Object::toString)
            .sorted()
            .collect(Collectors.toList());
```

Reply:
This is a formatting change, but follows the same function.  The `map(Object::toString)` method is necessary to convert each URL to a string.  Perhaps I am missing what you are suggesting.

Reply:
I'm suggestion adding `.sorted()` instead following with the `Collections.sort(sortedUrls);`.

Reply:
Thanks for the clarification, that makes sense, will make the change.